Source/LES.H

```diff
@@ -518,5 +518,591 @@ pc_dynamic_smagorinsky_sfs_term(
+  alpha = 2.0 * 0.5 * (q(i, j, k, QRHO) + q(i, j, k - 1, QRHO)) * deltabar *
+          deltabar * Sijmag * Sijmag;
+
+  const amrex::Real dTdz = dxinv * (q(i, j, k, QTEMP) - q(i, j, k - 1, QTEMP));
+  flux_T = mut * dTdz;
+}
+
+AMREX_GPU_DEVICE
+AMREX_FORCE_INLINE
+void
+pc_wale_sfs_term(
+  const int i,
+  const int j,
+  const int k,
+  const amrex::Array4<const amrex::Real>& q,
+  const amrex::Array4<const amrex::Real>& td,
+  const amrex::Array4<const amrex::Real>& a,
+  const amrex::Real dx,
+  const int dir,
+  const amrex::Real Cw,
+  const amrex::Real CI,
+  const amrex::Real PrT,
+  const amrex::Array4<amrex::Real>& flx)
+{
+
+  const amrex::IntVect iv{AMREX_D_DECL(i, j, k)};
+  const amrex::IntVect ivm(iv - amrex::IntVect::TheDimensionVector(dir));
+
+  // SFS stress
+  const amrex::Real dxinv = 1.0 / dx;
+  const amrex::Real deltabar = dx;
+  const amrex::Real Cw2 = Cw * Cw;
+  const amrex::Real third = 1.0 / 3.0;
+  amrex::Real alphaij[AMREX_SPACEDIM] = {0.0}, alpha, flux_T;
+  if (dir == 0) {
+    get_wale_sfs_stresses_xdir(
+      i, j, k, q, td, dxinv, deltabar, alphaij[0], alphaij[1], alphaij[2],
+      alpha, flux_T);
+  } else if (dir == 1) {
+    get_wale_sfs_stresses_ydir(
+      i, j, k, q, td, dxinv, deltabar, alphaij[0], alphaij[1], alphaij[2],
+      alpha, flux_T);
+  } else {
+    get_wale_sfs_stresses_zdir(
+      i, j, k, q, td, dxinv, deltabar, alphaij[0], alphaij[1], alphaij[2],
+      alpha, flux_T);
+  }
+  const amrex::Real sigmadx =
+    Cw2 * alphaij[0] - third * CI * alpha * static_cast<amrex::Real>(dir == 0);
+  const amrex::Real sigmady =
+    Cw2 * alphaij[1] - third * CI * alpha * static_cast<amrex::Real>(dir == 1);
+  const amrex::Real sigmadz =
+    Cw2 * alphaij[2] - third * CI * alpha * static_cast<amrex::Real>(dir == 2);
+  flx(iv, UMX) = -sigmadx;
+  flx(iv, UMY) = -sigmady;
+  flx(iv, UMZ) = -sigmadz;
+
+  // SFS turbulent diffusion
+  const amrex::Real Uface[AMREX_SPACEDIM] = {
+    static_cast<amrex::Real>(0.5 * (q(iv, QU) + q(ivm, QU))),
+    static_cast<amrex::Real>(0.5 * (q(iv, QV) + q(ivm, QV))),
+    static_cast<amrex::Real>(0.5 * (q(iv, QW) + q(ivm, QW)))};
+  flx(iv, UEDEN) =
+    -sigmadx * Uface[0] - sigmady * Uface[1] - sigmadz * Uface[2];
+
+  // SFS heat flux - move state from cell centers to faces to compute cp for
+  // flux at face
+  amrex::Real T = 0.5 * (q(iv, QTEMP) + q(ivm, QTEMP));
+  amrex::Real massfrac[NUM_SPECIES];
+  for (int n = 0; n < NUM_SPECIES; n++) {
+    massfrac[n] = 0.5 * (q(iv, QFS + n) + q(ivm, QFS + n));
+  }
+  amrex::Real cp = 0.0;
+  auto eos = pele::physics::PhysicsType::eos();
+  eos.TY2Cp(T, massfrac, cp);
+  flx(iv, UEDEN) = flx(iv, UEDEN) - cp * Cw2 / PrT * flux_T;
+
+  // Scale fluxes by area
+  flx(iv, UMX) = flx(iv, UMX) * a(iv);
+  flx(iv, UMY) = flx(iv, UMY) * a(iv);
+  flx(iv, UMZ) = flx(iv, UMZ) * a(iv);
+  flx(iv, UEDEN) = flx(iv, UEDEN) * a(iv);
+}
+
+AMREX_GPU_DEVICE
+AMREX_FORCE_INLINE
+void
+get_vreman_sfs_stresses_xdir(
```

Comment:
Can these be simplified to have one function that takes `const int dir` as an input rather than having three separate functions to reduce duplicate code? I know the separate function for x/y/z is used elsewhere in LES, so perhaps its ok to leave this for now and clean it all up later.